spring-integration-core/src/main/java/org/springframework/integration/util/ClassUtils.java

```diff
@@ -180,6 +188,7 @@ public static Class<?> findClosestMatch(Class<?> type, Set<Class<?>> candidates,
 				closestMatch = candidate;
 			}
 			else if (failOnTie && typeDiffWeight < Integer.MAX_VALUE && (typeDiffWeight == minTypeDiffWeight)) {
+				Objects.requireNonNull(closestMatch, "closestMatch should not be null at this point");
```

Comment:
Instead of this this do a ternary operator in the place of `closestMatch.getName()`.
There is a case when `if (typeDiffWeight < minTypeDiffWeight) {` is not true.

Reply:
I still believe `closestMatch` should not be null at this point, but I might be missing something.

To reach this `else if`, we need `typeDiffWeight == minTypeDiffWeight && typeDiffWeight < Integer.MAX_VALUE`, which would mean `minTypeDiffWeight` was already updated from its initial `Integer.MAX_VALUE` in a previous iteration where `closestMatch` was assigned.

Flow is a bit delicate though... 😅

Reply:
OK.
But it is better to avoid extra method call for nothing.
I believe the ternary operator is much cheaper than this `Objects.requireNonNull`